divoom/device.py

```diff
@@ -1,30 +1,51 @@
 import bluetooth
+import logging
+from enum import Enum
 from divoom.protocol import parse_reply, split_reply
 
-HELLO_BYTES = [0x0, 0x5, 0x48, 0x45, 0x4c, 0x4c, 0x4f, 0x0]
+logger = logging.getLogger(__name__)
+
+
+class Type(Enum):
+    DEFAULT = 0
+    DITOO_PRO = 1
+
+
+CONFIG_MAP = {
+    # port, hello bytes, size
+    Type.DEFAULT: (4, [0x0, 0x5, 0x48, 0x45, 0x4C, 0x4C, 0x4F, 0x0], 11),
+    Type.DITOO_PRO: (2, None, 16),
+}
+
+
 class Device:
-    def __init__(self, addr, timeout=1):
+    def __init__(self, addr, type_: Type = Type.DEFAULT, timeout=1):
```

Comment:
Breaking change. Maybe I should swap `type_` and `timeout`

Reply:
I think it's alright, not a super popular project :grin: